core/shared/src/main/scala/zio/prelude/Ord.scala

```diff
@@ -246,17 +192,18 @@ object Ord extends Lawful[Ord] {
    * first compare the values for reference equality and then compare the
    * values using the specified function.
    */
-  def make[A](ord: (A, A) => Ordering): Ord[A] =
-    (l, r) => ord(l, r)
+  def make[A](ord: (A, A) => Ordering): Ord[A] = new Ord[A] {
+    override protected def checkCompare(l: A, r: A): Ordering = ord(l, r)
+  }
 
   /**
    * Constructs an instance from an `ord` function and a `equal0` function.
    * Since this takes a separate `equal0`, short-circuiting the equality check (failing fast) is possible.
    */
-  def makeFrom[A](ord: (A, A) => Ordering, equal0: Equal[A]): Ord[A] =
+  def makeFrom[A](ord: (A, A) => Ordering, partialOrd: PartialOrd[A]): Ord[A] =
```

Comment:
This is weird. What happens if the partial ordering disagrees with the total ordering?

Reply:
Probably it makes more sense to accept `(A, A) => Option[Ordering]` and convert that to a partial ordering.